packages/wallet/src/InMemoryUtxoRepository.ts

```diff
@@ -75,11 +103,41 @@ export class InMemoryUtxoRepository implements UtxoRepository {
     return [...this.#utxoSet.values()];
   }
 
+  public get availableUtxos(): Schema.Utxo {
+    return this.allUtxos.filter((utxo) => !this.#lockedUtxoSet.has(utxo));
+  }
+
   public get rewards(): Schema.Lovelace | null {
     return this.#delegationAndRewards.rewards ?? null;
   }
 
   public get delegation(): Schema.PoolId | null {
     return this.#delegationAndRewards.delegate ?? null;
   }
+
+  async #onTransaction({ transaction, confirmed }: OnTransactionArgs) {
+    const utxoLockedByTx: Schema.Utxo = [];
+    const inputs = transaction.body().inputs();
+    for (let inputIdx = 0; inputIdx < inputs.len(); inputIdx++) {
+      const { txId, index } = cslToOgmios.txIn(inputs.get(inputIdx));
+      const utxo = this.allUtxos.find(([txIn]) => txIn.txId === txId && txIn.index === index)!;
+      this.#lockedUtxoSet.add(utxo);
+      utxoLockedByTx.push(utxo);
+    }
+    const unlock = (spent?: boolean) => {
+      for (const utxo of utxoLockedByTx) {
+        this.#lockedUtxoSet.delete(utxo);
+        spent && this.#utxoSet.delete(utxo);
+      }
+    };
+    try {
+      await confirmed;
+      unlock(true);
+    } catch (error) {
+      unlock(false);
+      if (!(error instanceof TransactionError) || error.reason !== TransactionFailure.Timeout) {
+        await this.emit('transactionUntracked', transaction).catch(this.#logger.error);
```

Comment:
One of the reasons I chose `emittery` for events is because it has nice types. It creates string literal types and changing this to a different string would be a TypeScript error. It also correctly infers type of event payload when emitting and subscribing. Do you still think an enum is needed?

Reply:
Enum would make renaming the event easier. Is there any other benefit you had in mind in this case?

Reply:
Right. It just adds more formality to the type, and possibly improves the generated doc